### PR TITLE
[traffic-gen-in-vm] Refactor VMI creation

### PR DESCRIPTION
**Pull request by orelmisan**
Currently, the two VMIs are built using the same basic spec, but with different parameters, such as:
1. VMI name.
2. Affinity settings.
3. ContainerDisk image name.
4. NIC{1..2} MAC and PCI addresses.

In order to enable the traffic gen VMI to receive TRex configuration files using a ConfigMap, the traffic gen VMI spec
should receive additional parameters (and be independent from the VMI under test).

Move all high-level VMI building logic to the `vmi.go` file under the `checkup` package.
Keep the low-level VMI spec building logic under the `checkup/vmi` package.

Add an identical random suffix to all created objects:
1. VMI under test.
2. Traffic gen VMI.
3. Traffic gen ConfigMap.

### CODE REVIEW

**Comment by orelmisan** [comment]
Tested against an OpenShift Virtualization 4.14 cluster.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/7f28215291e7f6526a84e7898e6aeb1c85af8b55..1b631cb15309af28d2c9b0ed9c1ca4ed957de577): Updated commit message.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/1b631cb15309af28d2c9b0ed9c1ca4ed957de577..b2c76d20e4b44c83bf755107ffad72bff7b73578): Refactored last commit so it will be better suited for the follow-up PR.

**Review by RamLavi — COMMENTED** [review]
maybe spec.go-->vmispec.go or something the like?

File: .golangci.yml [hunk 1]
```diff
@@ -3,8 +3,6 @@
 linters-settings:
   gofmt:
     simplify: false
-  dupl:
```
Comment (RamLavi) [inline .golangci.yml:6]: Can you insert this in a separate PR?

File: pkg/internal/checkup/vmi.go [hunk 2]
```diff
@@ -54,86 +54,70 @@ const (
...
-		ContainerDiskImage:              checkupConfig.VMContainerDiskImage,
-		NetworkAttachmentDefinitionName: checkupConfig.NetworkAttachmentDefinitionName,
-		NICEastMACAddress:               checkupConfig.DPDKEastMacAddress.String(),
-		NICEastPCIAddress:               config.VMIEastNICPCIAddress,
-		NICWestMACAddress:               checkupConfig.DPDKWestMacAddress.String(),
-		NICWestPCIAddress:               config.VMIWestNICPCIAddress,
-		Username:                        config.VMIUsername,
-		Password:                        config.VMIPassword,
-	}
+	vmiUnderTest := newBaseVMI(checkupConfig)
```
Comment (RamLavi) [inline pkg/internal/checkup/vmi.go]: It seems redundant to iterate over the options in two places (`vmi.New` and `applyOptionsToVMI`).
How about creating a function baseVMIOptions and then something like this?
```
baseOptions := baseVMIOptions(checkupConfig)
vmiunderTestCustomOptions := []vmi.Option{
		vmi.WithAffinity(Affinity(checkupConfig.DPDKNodeLabelSelector, checkupConfig.PodUID)),
		vmi.WithSRIOVInterface(eastNetworkName, checkupConfig.DPDKEastMacAddress.String(), config.VMIEastNICPCIAddress),
		vmi.WithSRIOVInterface(westNetworkName, checkupConfig.DPDKWestMacAddress.String(), config.VMIWestNICPCIAddress),
		vmi.WithContainerDisk(rootDiskName, checkupConfig.VMContainerDiskImage),
		vmi.WithCloudInitNoCloudVolume(cloudInitDiskName, CloudInit(config.VMIUsername, config.VMIPassword)
}

return vmi.New(RandomizeName(VMIUnderTestNamePrefix),baseOptions, vmiunderTestCustomOptions)
```

Comment (orelmisan) [reply pkg/internal/checkup/vmi.go]: Good idea.

File: .golangci.yml (hunk 1 above)
Comment (orelmisan) [reply .golangci.yml:6]: Sure.

Comment (orelmisan) [reply .golangci.yml]: After implementing your suggestion, this change is no longer needed.

File: pkg/internal/checkup/vmi.go (hunk 2 above)
Comment (orelmisan) [reply pkg/internal/checkup/vmi.go]: Done.
It really improved the code.
Thank you.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/b2c76d20e4b44c83bf755107ffad72bff7b73578..a0860202fa55b2ef163ea604e35067a1da6f452e): Answered review comments.

**Pull request merged**